### ISSUE TITLE
Master mutation test run failed

### ISSUE THREAD

**Issue by Ocramius** [Roave/BetterReflection#654]
See https://github.com/Roave/BetterReflection/runs/753692410

Possibily due to random execution order (intentional), we may have caught some global state issue

**Comment by Ocramius** [comment]
Another one: https://github.com/Roave/BetterReflection/runs/756048594

**Comment by tomasnorre** [comment]
It's not directly related to the topic, but is it on purpose that you don't execute the mutation test with more threads? 

I know that there are a lot of tests and therefore the mutation tests takes quite some time. 
I have tested locally with and without the threads' parameter. 

| Without threds | With threds=8 |
|---	|---	|
| 10m 46s | 6m 10s|

The GitHub Action Steps for mutation test is currently ~18 minutes so perhaps there could be a gain with the threads there as well.

**Comment by Ocramius** [comment]
It's not really a blocker to getting things done on this repo, so far.

IMO worth checking https://github.com/infection/infection/releases/tag/0.20.0 instead, which will lower mutation tests to a fraction of what they currently are.

**Comment by tomasnorre** [comment]
That's right. That would be an improvement with more "effect".

I'm thinking CI should be for fast feedback that's why I am addressing this.

**Comment by Ocramius** [comment]
Yes, well, there are a lot of "fast" checks (phpunit/psalm/phpcs) and then mutation testing will always take a while, and it will only get worse if we don't use more functional/minimalist coding approaches :-)

There's also https://github.com/Roave/infection-static-analysis-plugin, which may help reducing mutation test time even further.

**Comment by tomasnorre** [comment]
The roave/infection-static-analysis-plugin you use already. Or do I misunderstand you?

**Comment by Ocramius** [comment]
Oh wow, didn't realize we were already using it here

**Comment by tomasnorre** [comment]
I only saw it as I was looking for which check to run locally before committing.

**Comment by Ocramius** [comment]
Closing here: CI runs work again.
